### PR TITLE
feat(plasma-theme-builder): Add create and edit complex gradient to theme builder for different platforms

### PR DESCRIPTION
**Pull request by neretin-trike**
Обновлён React до 18 версии в веб-сервисе ThemBuilder и добавлены недостающие типы.

Удалён локальный шрифт `SBSansTextMono` и добавлена загрузка его из cdn.

Добавлено переключение режима темы (темный / светлый) веб-сервиса в зависимости от настроек системы.

Произведен масштабный рефакторинг утилитарных функций и типов. Теперь они лежат в соответствующих директориях `utils` и `types` и доступны для всего приложения.

Добавлена возможность создавать токены сложных градиентов для платформ android и ios с помощью UI в веб-сервисе.

:baby_chick: Download canary assets:

[color_asdk_ios-swift--canary.576.5411419114.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_asdk_ios-swift--canary.576.5411419114.swift)  
[color_asdk_kotlin--canary.576.5411419114.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_asdk_kotlin--canary.576.5411419114.kt)  
[color_asdk_react-native--canary.576.5411419114.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_asdk_react-native--canary.576.5411419114.ts)  
[color_asdk_xml--canary.576.5411419114.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_asdk_xml--canary.576.5411419114.xml)  
[color_plasma_b2c_ios-swift--canary.576.5411419114.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_plasma_b2c_ios-swift--canary.576.5411419114.swift)  
[color_plasma_b2c_kotlin--canary.576.5411419114.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_plasma_b2c_kotlin--canary.576.5411419114.kt)  
[color_plasma_b2c_react-native--canary.576.5411419114.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_plasma_b2c_react-native--canary.576.5411419114.ts)  
[color_plasma_b2c_xml--canary.576.5411419114.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_plasma_b2c_xml--canary.576.5411419114.xml)  
[color_sberHealth_ios-swift--canary.576.5411419114.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_ios-swift--canary.576.5411419114.swift)  
[color_sberHealth_kotlin--canary.576.5411419114.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_kotlin--canary.576.5411419114.kt)  
[color_sberHealth_react-native--canary.576.5411419114.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_react-native--canary.576.5411419114.ts)  
[color_sberHealth_xml--canary.576.5411419114.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_xml--canary.576.5411419114.xml)  
[color_sbermarket_business_ios-swift--canary.576.5411419114.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_ios-swift--canary.576.5411419114.swift)  
[color_sbermarket_business_kotlin--canary.576.5411419114.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_kotlin--canary.576.5411419114.kt)  
[color_sbermarket_business_react-native--canary.576.5411419114.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_react-native--canary.576.5411419114.ts)  
[color_sbermarket_business_xml--canary.576.5411419114.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_xml--canary.576.5411419114.xml)  
[color_sbermarket_ios-swift--canary.576.5411419114.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.576.5411419114.swift)  
[color_sbermarket_kotlin--canary.576.5411419114.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.576.5411419114.kt)  
[color_sbermarket_metro_ios-swift--canary.576.5411419114.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_ios-swift--canary.576.5411419114.swift)  
[color_sbermarket_metro_kotlin--canary.576.5411419114.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_kotlin--canary.576.5411419114.kt)  
[color_sbermarket_metro_react-native--canary.576.5411419114.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_react-native--canary.576.5411419114.ts)  
[color_sbermarket_metro_xml--canary.576.5411419114.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_xml--canary.576.5411419114.xml)  
[color_sbermarket_react-native--canary.576.5411419114.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.576.5411419114.ts)  
[color_sbermarket_selgros_ios-swift--canary.576.5411419114.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_ios-swift--canary.576.5411419114.swift)  
[color_sbermarket_selgros_kotlin--canary.576.5411419114.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_kotlin--canary.576.5411419114.kt)  
[color_sbermarket_selgros_react-native--canary.576.5411419114.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_react-native--canary.576.5411419114.ts)  
[color_sbermarket_selgros_xml--canary.576.5411419114.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_xml--canary.576.5411419114.xml)  
[color_sbermarket_wlbusiness_ios-swift--canary.576.5411419114.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_ios-swift--canary.576.5411419114.swift)  
[color_sbermarket_wlbusiness_kotlin--canary.576.5411419114.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_kotlin--canary.576.5411419114.kt)  
[color_sbermarket_wlbusiness_react-native--canary.576.5411419114.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_react-native--canary.576.5411419114.ts)  
[color_sbermarket_wlbusiness_xml--canary.576.5411419114.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_xml--canary.576.5411419114.xml)  
[color_sbermarket_xml--canary.576.5411419114.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_xml--canary.576.5411419114.xml)  
[color_sberonline_ios-swift--canary.576.5411419114.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_ios-swift--canary.576.5411419114.swift)  
[color_sberonline_kotlin--canary.576.5411419114.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_kotlin--canary.576.5411419114.kt)  
[color_sberonline_react-native--canary.576.5411419114.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_react-native--canary.576.5411419114.ts)  
[color_sberonline_xml--canary.576.5411419114.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_xml--canary.576.5411419114.xml)  
[color_sberprime_ios-swift--canary.576.5411419114.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.576.5411419114.swift)  
[color_sberprime_kotlin--canary.576.5411419114.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.576.5411419114.kt)  
[color_sberprime_react-native--canary.576.5411419114.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.576.5411419114.ts)  
[color_sberprime_xml--canary.576.5411419114.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_xml--canary.576.5411419114.xml)  
[shadow_sbermarket_react-native--canary.576.5411419114.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.576.5411419114.ts)  
[typo_mage_ios-swift--canary.576.5411419114.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.576.5411419114.swift)  
[typo_mage_kotlin--canary.576.5411419114.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.576.5411419114.kt)  
[typo_mage_react-native--canary.576.5411419114.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.576.5411419114.ts)  
[typo_plasma_ios-swift--canary.576.5411419114.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.576.5411419114.swift)  
[typo_plasma_kotlin--canary.576.5411419114.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.576.5411419114.kt)  
[typo_plasma_react-native--canary.576.5411419114.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.576.5411419114.ts)  
[typo_ruler_ios-swift--canary.576.5411419114.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.576.5411419114.swift)  
[typo_ruler_kotlin--canary.576.5411419114.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.576.5411419114.kt)  
[typo_ruler_react-native--canary.576.5411419114.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.576.5411419114.ts)  
[typo_sage_ios-swift--canary.576.5411419114.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.576.5411419114.swift)  
[typo_sage_kotlin--canary.576.5411419114.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.576.5411419114.kt)  
[typo_sage_react-native--canary.576.5411419114.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.576.5411419114.ts)  
[typo_sbermarket_ios-swift--canary.576.5411419114.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.576.5411419114.swift)  
[typo_sbermarket_kotlin--canary.576.5411419114.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.576.5411419114.kt)  
[typo_sbermarket_react-native--canary.576.5411419114.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.576.5411419114.ts)  
[typo_soulmate_ios-swift--canary.576.5411419114.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.576.5411419114.swift)  
[typo_soulmate_kotlin--canary.576.5411419114.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.576.5411419114.kt)  
[typo_soulmate_react-native--canary.576.5411419114.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.576.5411419114.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.218.0-canary.576.5411419114.0
  npm install @salutejs/cra-template-plasma-basic-template@2.4.0-canary.576.5411419114.0
  npm install @salutejs/plasma-colors@0.8.0-canary.576.5411419114.0
  npm install @salutejs/plasma-core@1.123.0-canary.576.5411419114.0
  npm install @salutejs/plasma-hope@1.218.0-canary.576.5411419114.0
  npm install @salutejs/plasma-icons@1.152.0-canary.576.5411419114.0
  npm install @salutejs/cra-template-plasma-shop-template@2.3.0-canary.576.5411419114.0
  npm install @salutejs/plasma-temple@1.167.0-canary.576.5411419114.0
  npm install @salutejs/plasma-tokens-b2b@1.23.0-canary.576.5411419114.0
  npm install @salutejs/plasma-tokens-b2c@0.32.0-canary.576.5411419114.0
  npm install @salutejs/plasma-tokens-core@0.3.0-canary.576.5411419114.0
  npm install @salutejs/plasma-tokens-web@1.38.0-canary.576.5411419114.0
  npm install @salutejs/plasma-tokens@1.57.0-canary.576.5411419114.0
  npm install @salutejs/plasma-typo@0.35.0-canary.576.5411419114.0
  npm install @salutejs/plasma-ui@1.199.0-canary.576.5411419114.0
  npm install @salutejs/plasma-web@1.218.0-canary.576.5411419114.0
  npm install @salutejs/plasma-cy-utils@0.64.0-canary.576.5411419114.0
  npm install @salutejs/plasma-sb-utils@0.121.0-canary.576.5411419114.0
  npm install @salutejs/plasma-tokens-utils@0.28.0-canary.576.5411419114.0
  # or 
  yarn add @salutejs/plasma-b2c@1.218.0-canary.576.5411419114.0
  yarn add @salutejs/cra-template-plasma-basic-template@2.4.0-canary.576.5411419114.0
  yarn add @salutejs/plasma-colors@0.8.0-canary.576.5411419114.0
  yarn add @salutejs/plasma-core@1.123.0-canary.576.5411419114.0
  yarn add @salutejs/plasma-hope@1.218.0-canary.576.5411419114.0
  yarn add @salutejs/plasma-icons@1.152.0-canary.576.5411419114.0
  yarn add @salutejs/cra-template-plasma-shop-template@2.3.0-canary.576.5411419114.0
  yarn add @salutejs/plasma-temple@1.167.0-canary.576.5411419114.0
  yarn add @salutejs/plasma-tokens-b2b@1.23.0-canary.576.5411419114.0
  yarn add @salutejs/plasma-tokens-b2c@0.32.0-canary.576.5411419114.0
  yarn add @salutejs/plasma-tokens-core@0.3.0-canary.576.5411419114.0
  yarn add @salutejs/plasma-tokens-web@1.38.0-canary.576.5411419114.0
  yarn add @salutejs/plasma-tokens@1.57.0-canary.576.5411419114.0
  yarn add @salutejs/plasma-typo@0.35.0-canary.576.5411419114.0
  yarn add @salutejs/plasma-ui@1.199.0-canary.576.5411419114.0
  yarn add @salutejs/plasma-web@1.218.0-canary.576.5411419114.0
  yarn add @salutejs/plasma-cy-utils@0.64.0-canary.576.5411419114.0
  yarn add @salutejs/plasma-sb-utils@0.121.0-canary.576.5411419114.0
  yarn add @salutejs/plasma-tokens-utils@0.28.0-canary.576.5411419114.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-576/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-576/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-576/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-576/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-576/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-576/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-576/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

File: website/plasma-theme-builder/src/GlobalStyle.tsx [hunk 1]
```diff
@@ -18,15 +19,29 @@ const DocumentStyle = createGlobalStyle`
...
-    <>
-        <DocumentStyle />
-        <ThemeStyle />
-        <TypoStyle />
-        <OldTypoStyle />
-    </>
-);
+const matchMediaWithColorScheme = window.matchMedia('(prefers-color-scheme: dark');
+
+const getSystemTheme = () => (matchMediaWithColorScheme.matches ? 'dark' : 'light' || 'dark');
```
Comment (Yakutoc) [inline website/plasma-theme-builder/src/GlobalStyle.tsx:29]: @neretin-trike Я тут сломался )) объясни пожалуйста )

File: website/plasma-theme-builder/src/components/PullRequest/PullRequest.tsx [hunk 2]
```diff
@@ -28,6 +28,14 @@ const Header = styled(H5)`
     margin-bottom: 1rem;
 `;
 
+const getFilesPath = (name?: string) => ({
+    theme: `packages/plasma-tokens/data/themes/${name}.json`,
```
Comment (Yakutoc) [inline website/plasma-theme-builder/src/components/PullRequest/PullRequest.tsx:32]: [Уточнение]

name - как необязательный аргумент. Что будет если прилетит nullable значение?

File: website/plasma-theme-builder/src/GlobalStyle.tsx (hunk 1 above)
Comment (neretin-trike) [reply website/plasma-theme-builder/src/GlobalStyle.tsx:29]: да тут не правильная логика, спасиб 👍

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-576/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-576/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-576/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-576/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-576/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-576/temple-storybook/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-576/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-576/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-576/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-576/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-576/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-576/temple-storybook/

**Review by Yakutoc — APPROVED** [review]

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-576/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-576/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-576/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-576/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-576/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-576/temple-storybook/

File: website/plasma-theme-builder/src/GlobalStyle.tsx [hunk 3]
```diff
@@ -1,27 +1,47 @@
...
-    </>
-);
+const matchMediaWithColorScheme = window.matchMedia('(prefers-color-scheme: dark');
+
+const getSystemTheme = () => (matchMediaWithColorScheme.matches ? 'dark' : 'light');
+
+export const GlobalStyle = () => {
+    const [theme, setTheme] = useState<ThemeMode>(getSystemTheme());
+
+    matchMediaWithColorScheme.addEventListener('change', ({ matches }) => {
```
Comment (Yeti-or) [inline website/plasma-theme-builder/src/GlobalStyle.tsx:34]: вот интересно а не можем ли мы это убрать внутрь самих тем?

File: website/plasma-theme-builder/src/components/GradientTokenSwiftLayer/GradientTokenSwiftLayer.tsx [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+}
+
+export const GradientTokenSwiftLayer = ({
+    originalColor,
+    swift,
+    index,
+    onChangeSwiftValue,
+}: GradientTokenSwiftLayerProps) => {
+    const onChangeValue = useCallback(
+        (event: ChangeEvent<HTMLInputElement>) => {
```
Comment (Yeti-or) [inline website/plasma-theme-builder/src/components/GradientTokenSwiftLayer/GradientTokenSwiftLayer.tsx:74]: а кстати вот так не работает? `useCallback<ChangeEventHandler<HTMLInputElement>>((event) => ...)`

File: website/plasma-theme-builder/src/components/SolidTokenValue/SolidTokenValue.tsx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+import { getAccentColors, getSaturations } from '../../utils';
+import type { GeneralColor, InputDataValue } from '../../types';
+
+const Root = styled.div``;
+
+const StyledSelect = styled(Select)`
+    /* stylelint-disable-next-line selector-max-universal */
+    * {
+        /* stylelint-disable-next-line font-weight-notation */
+        font-weight: 400;
```
Comment (Yeti-or) [inline website/plasma-theme-builder/src/components/SolidTokenValue/SolidTokenValue.tsx:17]: выглядит как у нас чего-то не хватает

File: website/plasma-theme-builder/src/components/mixins.ts [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import { css } from 'styled-components';
 
 export const SBSansTextMono = css`
-    font-family: 'SB Sans Text Mono';
+    font-family: 'SB Sans Text Mono', Helvetica, Arial, sans-serif;
```
Comment (Yeti-or) [inline website/plasma-theme-builder/src/components/mixins.ts:4]: Хотелось бы вообще понять что это за шрифт и почему он у нас только в theme-builder, рассскажите на каком-нибудь дизайн синке

File: website/plasma-theme-builder/src/index.tsx [hunk 7]
```diff
@@ -1,13 +1,20 @@
...
+}
+
+const root = createRoot(rootElement);
+
+root.render(
+    <React.StrictMode>
         <App />
-    </>,
-    document.getElementById('root'),
+        <GlobalStyle />
```
Comment (Yeti-or) [inline website/plasma-theme-builder/src/index.tsx:18]: а почему они после App ??

File: website/plasma-theme-builder/src/utils/other.tsx [hunk 8]
```diff
@@ -0,0 +1,25 @@
+import { nanoid } from 'nanoid';
+
+export const getUUID = () => {
+    const value = window.localStorage.getItem('uuid');
+
+    if (value) {
+        return value;
+    }
+
+    const pluginClientId = nanoid();
```
Comment (Yeti-or) [inline website/plasma-theme-builder/src/utils/other.tsx:10]: ну в целом можно и Crypto заюзать
```
window.crypto.randomUUID();
```

**Review by Yeti-or — APPROVED** [review]

File: website/plasma-theme-builder/src/GlobalStyle.tsx (hunk 3 above)
Comment (neretin-trike) [reply website/plasma-theme-builder/src/GlobalStyle.tsx:34]: ну сейчас мы отдаём пользователям весь набор наших тем (и светлых и тёмных), а они в свою очередь могут подключать что угодно угодно (хоть только темную) и в таком случае механизм смены темы зашить внутрь не получится.
Но в целом, в будущем, например, мы можем отдавать стиль содержащий полный набор (и светлой и темной темы т.е.), который можно будет переключать или по-какому-нибудь свичу, или как здесь - по системным настройкам.

И будет выглядеть как-нибудь типа:

```ts
import { getGlobalTheme } from '@salutejs/plasma-tokens';

const Theme = getGlobalTheme('sbermarket_metro');

export const GlobalStyle = () => {
    const themeType: 'dark' | 'light' | 'system' = 'system';

    return (
        <>
            <Theme type={themeType}/>
            <TypoStyle />
        </>
    );
};

```

File: website/plasma-theme-builder/src/components/GradientTokenSwiftLayer/GradientTokenSwiftLayer.tsx (hunk 4 above)
Comment (neretin-trike) [reply website/plasma-theme-builder/src/components/GradientTokenSwiftLayer/GradientTokenSwiftLayer.tsx:74]: так тоже работает. Но кажется, что у нас в проекте в большинстве случае объявление типа идёт внутри. Можем договориться и привести какому-то единообразию

File: website/plasma-theme-builder/src/components/SolidTokenValue/SolidTokenValue.tsx (hunk 5 above)
Comment (neretin-trike) [reply website/plasma-theme-builder/src/components/SolidTokenValue/SolidTokenValue.tsx:17]: ага, это немного странный кейс. Для самих шрифтов мы можем указывать жирность (`bold`). Но поскольку шрифт меняется внутри Select, то и управлять я им не могу

File: website/plasma-theme-builder/src/components/mixins.ts (hunk 6 above)
Comment (neretin-trike) [reply website/plasma-theme-builder/src/components/mixins.ts:4]: оке 👌 Вообще как я понял это наш "корпоративный шрифт" для пазмы. В наших архетипах он так и называется - `plasma`

File: website/plasma-theme-builder/src/index.tsx (hunk 7 above)
Comment (neretin-trike) [reply website/plasma-theme-builder/src/index.tsx:18]: кажется, что разницы нет, но давай на всякий случай переставлю

**Review by kayman233 — APPROVED** [review]

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-576/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-576/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-576/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-576/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-576/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-576/temple-storybook/

**Pull request merged**